### PR TITLE
feat(misconf): support for ignore by nested attributes

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description

Added ability to ignore by nested attributes and values.

## Related issues
- Part of https://github.com/aquasecurity/trivy/issues/7180

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/iac/scanners/terraform/ignore_test.go [hunk 1]
```diff
@@ -433,7 +433,7 @@ resource "bad" "my-rule" {
 		{
 			name: "ignore by dynamic block value",
 			inputOptions: `
-// trivy:ignore:*[secure_settings.enabled=false]
+// trivy:ignore:*[secure_settings.0.enabled=false]
```
Comment (simar7) [inline pkg/iac/scanners/terraform/ignore_test.go:436]: Just wondering if you don't pass the index in now like so, does it break existing behaviour (backwards compatibility)?

Comment (nikpivkin) [reply pkg/iac/scanners/terraform/ignore_test.go]: Now in Trivy if a resource has multiple blocks, only the first one is used for ignoring. [Kept](https://github.com/aquasecurity/trivy/pull/7205/commits/86e8b29d5d15c8bec427de9535e0051a26b7346f) the previous behaviour for backwards compatibility.

File: pkg/iac/scanners/terraform/ignore_test.go [hunk 2]
```diff
@@ -433,7 +433,7 @@ resource "bad" "my-rule" {
 		{
 			name: "ignore by dynamic block value",
 			inputOptions: `
-// trivy:ignore:*[secure_settings.0.enabled=false]
+// trivy:ignore:*[secure_settings.enabled=false]
```
Comment (simar7) [inline pkg/iac/scanners/terraform/ignore_test.go:436]: let's add these two as separate test cases so they're both documented in behavior.

Comment (nikpivkin) [reply pkg/iac/scanners/terraform/ignore_test.go]: Done https://github.com/aquasecurity/trivy/pull/7205/commits/d71e6ac6bbc462df34e31e36debe6a903f534616

**Review by simar7 — APPROVED** [review]

**Pull request merged**